Stack frame size exceeded when building with clang-15

```
cmake . -DCMAKE_BUILD_TYPE="DEBUG" -DCMAKE_C_COMPILER="clang-15" -DCMAKE_CXX_COMPILER="clang++-15" -DDYNINST_WARNINGS_AS_ERRORS=ON -DDYNINST_CXX_LANGUAGE_STANDARD=11

...

[ 14%] Building CXX object instructionAPI/CMakeFiles/instructionAPI.dir/src/InstructionDecoder-power.C.o
In file included from /__w/dyninst/dyninst/instructionAPI/src/InstructionDecoder-power.C:915:
/__w/dyninst/dyninst/instructionAPI/src/power_opcode_tables.C:57:34: error: stack frame size (37512) exceeds limit (30000) in 'Dyninst::InstructionAPI::power_entry::buildTables()::$_0::operator()() const' [-Werror,-Wframe-larger-than]
    std::call_once(built_tables, [&]() {
                                 ^
1 error generated.
gmake[2]: *** [instructionAPI/CMakeFiles/instructionAPI.dir/build.make:258: instructionAPI/CMakeFiles/instructionAPI.dir/src/InstructionDecoder-power.C.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:413: instructionAPI/CMakeFiles/instructionAPI.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
```

I didn't see this with clang <= 14 and it's not a problem in non-debug builds.

CI logs: https://github.com/dyninst/dyninst/actions/runs/4822439096/jobs/8589676436#step:7:71

--- a/cmake/DyninstWarnings.cmake
+++ b/cmake/DyninstWarnings.cmake
@@ -163,7 +163,7 @@ if(HAS_CPP_FLAG_Wframe_larger_than AND NOT DYNINST_DISABLE_DIAGNOSTIC_SUPPRESSIO
       set(nonDebugMaxFrameSizeOverrideFinalizeOperands 30000)
     endif()
   elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
-    if(${CMAKE_CXX_COMPILER_VERSION} MATCHES "^14(\.|$)")
+    if(${CMAKE_CXX_COMPILER_VERSION} MATCHES "^1[4-6](\.|$)")
       set(debugMaxFrameSizeOverridePowerOpcodeTable 40000)
     else()
       set(debugMaxFrameSizeOverridePowerOpcodeTable 30000)

